### ISSUE TITLE
Accessories show/controllable  in homebridge, no response in Home

### ISSUE THREAD

**Issue by russellkmoore** [hjdhjd/homebridge-ratgdo#9]
### Homebridge Ratgdo

2.2.0

### Homebridge Platform and OS

Raspbian GNU/Linux Buster (10)

### Homebridge Version

v1.8.2

### Node Version

 v18.18.2

### Ratgdo ESPHome Firmware Version

2024.5.0 

### Describe the problem

New install. was able to successfully put firmware on 2 separate 2.53i boards (for 2 doors, both Liftmaster 8500W). Successfully installed Homebridge RATGDO plugin, set up child bridge. Accessories showed up in Hombridge Accessory panel. Successfully added child bridge to Home app, and it saw the 2 garage door openers. Set Both Up, Separated Tiles to show Garage Door, Light, Motion Sensor, and Switch for each. All show "no response" from the get-go in Home. In Homebridge, all are reporting in and controllable as expected. Have tried reinstalling plugin, restarting bridge, removing and re-adding bridge to home, and waiting several minutes with no success.

<img width="1774" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/710701e4-b16c-42b3-a671-64011fb5e6d8">

<img width="1197" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/ee9cda90-4cf5-46e4-85dd-787563b23d89">

<img width="519" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/48fac5e9-ad3b-41f6-a527-faa88960ed19">

<img width="515" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/e4994fcc-ca12-4245-af93-39b9609865a0">

### Homebridge HBR JSON configuration

```json
{
    "name": "Ratgdo",
    "platform": "Ratgdo",
    "options": [
        "Enable.Opener.Switch"
    ],
    "_bridge": {
        "username": "0E:81:ED:90:3C:57",
        "port": 35137
    }
}
```

### Relevant log output

```shell
child bridge logs upon restart:


[21/05/2024, 19:59:37] [Ratgdo] Restarting child bridge...
[21/05/2024, 19:59:37] Got SIGTERM, shutting down child bridge process...
[21/05/2024, 19:59:42] [Ratgdo] Child bridge process ended
[21/05/2024, 19:59:42] [Ratgdo] Process Ended. Code: 143, Signal: null
[21/05/2024, 19:59:49] [Ratgdo] Restarting Process...
[21/05/2024, 19:59:50] [Ratgdo] Launched child bridge with PID 1724
[21/05/2024, 19:59:51] Registering platform 'homebridge-ratgdo.Ratgdo'
[21/05/2024, 19:59:51] [Ratgdo] Loaded homebridge-ratgdo v2.2.0 child bridge successfully
[21/05/2024, 19:59:51] Loaded 2 cached accessories from cachedAccessories.0E81ED903C57.
[21/05/2024, 19:59:51] Publishing bridge accessory (name: Ratgdo, publishInfo: {
  username: '0E:81:ED:90:3C:57',
  port: 35137,
  pincode: '***-**-***',
  category: 2,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'avahi'
}).
[21/05/2024, 19:59:51] [Ratgdo] Discovered: ratgdov2.5i 4b0a76 (address: 192.168.0.44 mac: 08F9E04B0A76 ESPHome firmware: v2024.5.0).
[21/05/2024, 19:59:51] [Ratgdo] Configuring: ratgdov2.5i 4b0a76 (address: 192.168.0.44 mac: 08F9E04B0A76 ESPHome firmware: v2024.5.0).
[21/05/2024, 19:59:51] [Ratgdo] Small Garage Door: Enabling automation door opener switch.
[21/05/2024, 19:59:51] [Ratgdo] Discovered: ratgdov2.5i 4b0a6e (address: 192.168.0.36 mac: 08F9E04B0A6E ESPHome firmware: v2024.5.0).
[21/05/2024, 19:59:51] [Ratgdo] Configuring: ratgdov2.5i 4b0a6e (address: 192.168.0.36 mac: 08F9E04B0A6E ESPHome firmware: v2024.5.0).
[21/05/2024, 19:59:51] [Ratgdo] Large Garage Door: Enabling automation door opener switch.
[21/05/2024, 19:59:51] Homebridge v1.8.2 (HAP v0.12.1) (Ratgdo) is running on port 35137.
[21/05/2024, 19:59:51] [Ratgdo] Small Garage Door: Ratgdo connected.
[21/05/2024, 19:59:51] [Ratgdo] Large Garage Door: Ratgdo connected.



Then went into Accessories Panel on Homebridge and clicked on Small Garage Door:

Garage Door opende, Tiles changed state appropriately, and these lines were logged in plugin:

[21/05/2024, 20:01:11] [Ratgdo] Small Garage Door: Light on.
[21/05/2024, 20:01:11] [Ratgdo] Small Garage Door: Opening.
[21/05/2024, 20:01:23] [Ratgdo] Small Garage Door: Open.


Logs from RATGDO board unremarkable - reflect opening the door, turning light on etc.
```

### Acknowledgment that you are only running non-beta firmware releases from Apple, ESPHome, or Ratgdo

- [X] I acknowledge that I am not running any beta firmwares or operating systems from Apple, ESPHome, or Ratgdo.

**Comment by hjdhjd** [comment]
Thanks for the report. Everything appears correct.  HBR’s doing what it should.

The problem is the ESPHome 2024.5.0 firmware is quite buggy and was just released…you seem to have been caught up in it. If you’re comfortable with all-things-command-line I would recommend reflashing your Ratgdos with the prior ESPHome firmware 2024.4.2. Your best bet to do that is to download the [Docker ESPHome 2024.4.2 image](https://hub.docker.com/layers/esphome/esphome/2024.4.2/images/sha256-02fa50ad0d1fe39a63204b9da53e47e5a1455861602328ad4bf8c097130f6706?context=explore) and install it.

You can use the YAML on the HBR home page, or you can grab it off the Ratgdo homepage if you prefer - mine’s a bit more tailored and specific, but either work great.

[Docker command reference for ESPHome here](https://esphome.io/guides/faq.html?highlight=reboot#docker-reference).

[Bug report 5793 on the ESPHome](https://github.com/esphome/issues/issues/5793)

Best of luck…generally speaking ESPHome is rock solid…this is just an unfortunate regression on their part. I also wouldn’t advise upgrading a whole lot either…it’s a very simple device that you only need to do one thing correctly. The rest, HBR deals with. 😄

**Comment by russellkmoore** [comment]
Hoping my notes help anyone else who runs into this.

I've flashed with the 2024.4.2 firmware. 
_Note that for both of my devices, I had to run the "compile and upload" step twice. First time it would error during upload at about 12% and throw "IndexError: list index out of range" - just run it again and it completes_

<img width="1449" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/a4fe9ec1-c931-402c-816d-d2272ae6239f">



<img width="1449" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/a9a1e17f-3e40-4248-83ba-b252c8aa1144">

I agree, your YAML is nice at least to be able to set the names.
_Note that the hostname should match where you have set it previously (at least on Mac) so the .local resolution works. For me I believe this was cached in my UniFi Network setup (see "hostname" entry in detail panel - needs to match what you put in yaml file - e.g. name: "small-garage-door-controller"):_

<img width="1785" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/aa4fc5e5-61c0-48f2-93c0-7f7d53bd5ccd">

Clicking on actions in Homebridge produced immediate log lines in the device admin UI, and everything still reflected and works in homebridge UI as it did before:
<img width="1772" alt="image" src="https://github.com/hjdhjd/homebridge-ratgdo/assets/5854777/bdd46fe9-4b3f-44fa-9c17-9d5cc5d7efde">

**HOWEVER**
Initially it still showed as "No Response" in Home app on phone, nothing was controllable. 
Then opened Home app on desktop and it no longer showed "No Response" but did show incorrect state, and was also not controllable.

_Power cycled the AppleTV that was acting as the home hub. Within a minute, this seemed to fix everything everywhere._ 

I don't understand how the buggy firmware affected the state and controls pushed between Homebridge and Home, but apparently you are correct and this fixed it.

Thank you so much for your help!  I abandoned the HomeKit connection versions linked to from the RATGDO GitHub page because after a few days the device would just become non-responsive, and would usually require a re-flash to get connected again. Hoping that was all due to the firmware issues in esphome.

**Comment by hjdhjd** [comment]
I suspect so. Also...switching between ESPHome and non-ESPHome firmwares can do goofy things to the flash layout.

TL;DR: ESPHome is tried and true (2024.5.x notwithstanding). You shouldn't need to upgrade/update the firmware unless you are OCD (sadly I am...), but be prepared to roll back if needed. 99% of what HBR does not need anything special and really, Ratgdo is the kind of thing that you want to get up and running and never look at the firmware/environment again...and by and large you can do that.

HBR is where the "magic" as it were lives that provides some of the more feature-rich capabilities (occupancy, etc). My philosophy is - keep the Ratgdo as simple as possible and don't overcomplicate it. I just need it to expose door control, light control, motion, etc...I deal with the heavy lifting in HBR to make it all fancy.

It's been rock solid for me for quite some time since pivoting HBR over to the ESPHome firmware. It's nice to have a solution that "just works". 😄 

Glad you're up and running!
